### PR TITLE
feat: better merge queue messages

### PR DESCRIPTION
**Pull request by sileht**

### CODE REVIEW

**Comment by sileht** [comment]
@jd if you can review the messages before I record all tests

File: mergify_engine/actions/merge.py [hunk 1]
```diff
@@ -580,34 +579,47 @@ def get_queue_summary(ctxt):
...
-        if need_update:
-            title = "Base branch will be updated soon"
-            summary = (
-                "The pull request base branch will be updated soon and then merged."
-            )
+
+        summary = ""
+        if self.config["strict"] in ("smart+fasttrack", "smart+ordered"):
+            if self.config["strict"] == "smart+fasttrack" or queue.is_first_pull(ctxt):
+                title = "The pull request is queued for merge and the next to be merged"
```
Comment (jd) [inline mergify_engine/actions/merge.py:593]:
```suggestion
                title = "The pull request is the first in the queue to be merged"
```

File: mergify_engine/actions/merge.py [hunk 2]
```diff
@@ -580,34 +579,47 @@ def get_queue_summary(ctxt):
...
+            else:
+                title = "The pull request is queued for merge"
```
Comment (jd) [inline mergify_engine/actions/merge.py:595]: can we know the position?

File: mergify_engine/actions/merge.py [hunk 3]
```diff
@@ -580,34 +579,47 @@ def get_queue_summary(ctxt):
...
+
+            if is_behind:
+                summary = (
+                    "\nThe pull request base branch will be updated before get merged."
```
Comment (jd) [inline mergify_engine/actions/merge.py:599]:
```suggestion
                    "\nThe pull request base branch will be updated before being merged."
```

File: mergify_engine/actions/merge.py [hunk 4]
```diff
@@ -580,34 +579,47 @@ def get_queue_summary(ctxt):
...
+                )
+
+        elif self.config["strict"]:
+            if is_behind:
+                title = (
+                    "The pull request will be updated to follow its base branch soon"
```
Comment (jd) [inline mergify_engine/actions/merge.py:605]:
```suggestion
                    "The pull request will be updated with its base branch soon"
```

**Review by jd — CHANGES_REQUESTED** [review]
Let me know if you think this is ok.

File: mergify_engine/actions/merge.py (hunk 2 above)
Comment (sileht) [reply mergify_engine/actions/merge.py:595]: Yes

Comment (jd) [reply mergify_engine/actions/merge.py:595]: Ok then it might be good to add it ;)

File: mergify_engine/utils.py [hunk 5]
```diff
@@ -197,3 +197,12 @@ def get_random_choices(
                 break
 
     return picked
+
+
+ORDINAL_SUFFIXES = {1: "st", 2: "nd", 3: "rd"}
+
+
+def to_ordinal_numeric(number: int) -> str:
```
Comment (jd) [inline mergify_engine/utils.py:205]: add unit tests?

File: mergify_engine/actions/merge.py [hunk 6]
```diff
@@ -546,9 +547,8 @@ def merge_report(
         return check_api.Result(conclusion, title, summary)
 
     @staticmethod
-    def get_queue_summary(ctxt):
-        q = queue.Queue.from_context(ctxt)
-        pulls = q.get_pulls()
+    def get_queue_summary(ctxt: context.Context, queue: queue.Queue) -> str:
```
Comment (jd) [inline mergify_engine/actions/merge.py:550]:
```suggestion
    def get_queue_summary(ctxt: context.Context, q: queue.Queue) -> str:
```
or whatever that does not shadow the `queue` module

**Review by jd — APPROVED** [review]

File: mergify_engine/actions/merge.py (hunk 6 above)
Comment (sileht) [reply mergify_engine/actions/merge.py:550]: done

**Pull request merged**